docs/contents/user-docs/irsa.md

```diff
@@ -31,8 +31,150 @@ to assume the IAM Role with reduced permissions.
+By modifying the variables values as needed, these steps can be applied for other ACK controllers.
+
+The steps include:
+
+### 1. Create OIDC identity provider for cluster**
+Create OIDC identity provider for cluster using CLI command.
+Example:
+```
+EKS_CLUSTER_NAME=<eks cluster name>
+eksctl utils associate-iam-oidc-provider --cluster $EKS_CLUSTER_NAME --approve
+```
+For detailed instructions, follow [Enabling IAM roles for service accounts on your cluster][3].
+
+### 2. Create an IAM role and policy for service account**
+For detailed instructions, follow instructions at [Creating an IAM role and policy for your service account][4].
+
+#### 2(a) - Create IAM role
+```
+AWS_ACCOUNT_ID=$(aws sts get-caller-identity --query "Account" --output text)
+OIDC_PROVIDER=$(aws eks describe-cluster --name $EKS_CLUSTER_NAME --query "cluster.identity.oidc.issuer" --output text | sed -e "s/^https:\/\///")
+ACK_K8S_NAMESPACE=<Kubernetes namespace to use>
+ACK_K8S_SERVICE_ACCOUNT_NAME=ack-elasticache-controller
+
+read -r -d '' TRUST_RELATIONSHIP <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Federated": "arn:aws:iam::${AWS_ACCOUNT_ID}:oidc-provider/${OIDC_PROVIDER}"
+      },
+      "Action": "sts:AssumeRoleWithWebIdentity",
+      "Condition": {
+        "StringEquals": {
+          "${OIDC_PROVIDER}:sub": "system:serviceaccount:${ACK_K8S_NAMESPACE}:${ACK_K8S_SERVICE_ACCOUNT_NAME}"
+        }
+      }
+    }
+  ]
+}
+EOF
+echo "${TRUST_RELATIONSHIP}" > trust.json
+
+# update variables as needed
+AWS_SERVICE_NAME='elasticache'
+ACK_EKS_IRSA_IAM_ROLE_NAME="ACK-EKS-IRSA-${AWS_SERVICE_NAME}"
+ACK_EKS_IRSA_IAM_ROLE_DESCRIPTION='IRSA role for ACK $AWS_SERVICE_NAME controller deployment on EKS cluster using helm charts'
+aws iam create-role --role-name "${ACK_EKS_IRSA_IAM_ROLE_NAME}" --assume-role-policy-document file://trust.json --description "${ACK_EKS_IRSA_IAM_ROLE_DESCRIPTION}"
+
+```
+
+#### 2(b) - Attach IAM policy to role
+```
+# This example uses pre-existing policy for ElastiCache
+# Create an IAM policy and use its ARN and update IAM_POLICY_ARN variable as needed
+IAM_POLICY_ARN='arn:aws:iam::aws:policy/AmazonElastiCacheFullAccess'
+aws iam attach-role-policy \
+    --role-name "${ACK_EKS_IRSA_IAM_ROLE_NAME}" \
+    --policy-arn "$IAM_POLICY_ARN"
+```
+
+### 3. Associate an IAM role to service account**
+
+For detailed instructions, follow instructions at [Associate an IAM role to a service account ][5].
+
+#### 3(a) - If helm charts available on local file system
+
+Update `values.yaml` and set value for `aws.region`, `serviceAccount.annotations`.
+
+```
+# update variables as needed
+ACK_CONTROLLER_HELM_CHARTS_DIR=<directory containing helm chart for ACK service controller>
+AWS_SERVICE_NAME='elasticache'
+ACK_K8S_NAMESPACE=ack-system-test-helm
+ACK_K8S_DEPLOYMENT_NAME=ack-$AWS_SERVICE_NAME-controller-helm-test
```

Comment:
replace use of the word `DEPLOYMENT` above with `RELEASE`. The reason: `Deployment` is a Kubernetes construct and it's confusing since below what you are referring to is the Helm release.